src/ado2gh/Program.cs

```diff
@@ -23,15 +25,40 @@ public static async Task Main(string[] args)
                 .AddSingleton<AdoApiFactory>()
                 .AddSingleton<GithubApiFactory>()
                 .AddSingleton<RetryPolicy>()
+                .AddSingleton<VersionChecker>()
                 .AddHttpClient();
 
             var serviceProvider = serviceCollection.BuildServiceProvider();
-
             var parser = BuildParser(serviceProvider);
 
+            try
+            {
+                await LatestVersionCheck();
+            }
+            catch (Exception ex)
+            {
+                Logger.LogWarning($"Could not retrieve latest ado2gh version from github.com, please ensure you are using the latest version by downloading it from https://github.com/github/gh-gei/releases/latest");
+                Logger.LogVerbose(ex.ToString());
+            }
+
             await parser.InvokeAsync(args);
         }
 
+        private static async Task LatestVersionCheck()
+        {
+            var versionChecker = new VersionChecker();
```

Comment:
It's better to instantiate using the `serviceProvider`, pass it as a parameter the call `GerRequriedService`